utils/ecl-diff/ecl-diff-component.js

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env node
+
+/* eslint-disable no-console, no-param-reassign */
+
+const prettier = require('prettier');
+const fs = require('fs');
+const eclDiffVariant = require('./ecl-diff-variant.js');
+
+const rootFolder = process.cwd();
+const distFolder = `${rootFolder}/php`;
+// We retrieve all the available variants files (php ones).
+const getData = (component, system) => {
+  const systemFolder = `${distFolder}/packages/${system}`;
+  const twigFullPath = `${systemFolder}/${component}`;
+  const twigFilesFolder = fs.readdirSync(`${twigFullPath}`);
+  const twigFiles = twigFilesFolder.filter((elm) => {
+    return elm.match(/.*\.(php.html)/gi);
+  });
+
+  const variants = [];
+  twigFiles.forEach(async (twigFile) => {
+    const variant = twigFile.includes('--')
+      ? twigFile.replace(`${component}--`, '').slice(0, -9)
+      : 'default';
+    variants.push({ variant, file: twigFile, component });
+  });
+
+  return variants;
+};
+
+module.exports = (component, system) => {
+  const systemFolder = `${distFolder}/packages/${system}`;
+  const twigFullPath = `${systemFolder}/${component}`;
+  // This is for the console. Do not consider the same message in
+  // ecl-diff-variant.js a duplicate, then.
+  let message = `\nChecking component: ${component} - (${system})\n`;
+  message += `-------------------------------------------------------`;
+  console.log(message);
+  const datas = getData(component, system);
+  // Here we pass all the variants found to the eclDiffVariant promise in a
+  // Promise.all.
+  return Promise.all(
+    datas.map((variant) => eclDiffVariant(variant, system))
+  ).then((res) => {
+    // Update the story file with the ecl diff report in a storybook panel.
+    const diffPath = `${twigFullPath}/ecl-diff/${component}.md`;
+    if (fs.existsSync(diffPath)) {
+      fs.readFile(diffPath, 'utf8', (err, markdown) => {
```

Comment:
Mixing coding styles which gives a callback hell. Earlier in the same file you've tried async/await. If you go for this style, which is the modern one, remove the then and `const res = await Promise.all(...)`. Then the CPS-s of `readFile` could be done way better. You could either promisify the function which many have been doing in the migration of CPS to promises https://dev.to/sarneeh/comment/hhbl or even use the native promise version https://nodejs.org/api/fs.html#fs_fspromises_readfile_path_options

Reply:
Error handling: use `try/catch`